controllers/zookeepercluster_controller.go

```diff
@@ -671,30 +674,30 @@ func (r *ZookeeperClusterReconciler) yamlConfigMap(instance *zookeeperv1beta1.Zo
 	return yamlexporter.GenerateOutputYAMLFile(subdir, cm.Kind, cm)
 }
 
-func (r *ZookeeperClusterReconciler) reconcileFinalizers(instance *zookeeperv1beta1.ZookeeperCluster) (err error) {
+func (r *ZookeeperClusterReconciler) reconcileFinalizers(instance *zookeeperv1beta1.ZookeeperCluster) (finalized bool, err error) {
 	if instance.Spec.Persistence != nil && instance.Spec.Persistence.VolumeReclaimPolicy != zookeeperv1beta1.VolumeReclaimPolicyDelete {
-		return nil
+		return false, nil
 	}
 	if instance.DeletionTimestamp.IsZero() {
 		if !utils.ContainsString(instance.ObjectMeta.Finalizers, utils.ZkFinalizer) && !config.DisableFinalizer {
 			instance.ObjectMeta.Finalizers = append(instance.ObjectMeta.Finalizers, utils.ZkFinalizer)
 			if err = r.Client.Update(context.TODO(), instance); err != nil {
-				return err
+				return false, err
 			}
 		}
-		return r.cleanupOrphanPVCs(instance)
+		return false, r.cleanupOrphanPVCs(instance)
 	} else {
 		if utils.ContainsString(instance.ObjectMeta.Finalizers, utils.ZkFinalizer) {
 			if err = r.cleanUpAllPVCs(instance); err != nil {
-				return err
+				return false, err
 			}
 			instance.ObjectMeta.Finalizers = utils.RemoveString(instance.ObjectMeta.Finalizers, utils.ZkFinalizer)
 			if err = r.Client.Update(context.TODO(), instance); err != nil {
-				return err
+				return false, err
 			}
 		}
 	}
-	return nil
+	return true, nil
```

Comment:
Would be cleaner to return some kind of error at this point instead of introducing a boolean.